apache/iotdb
# [IOTDB-6138] Fix the support of negative timestamp

## Description

The current support of negative timestamp is not perfect. There are two main issues.

* When query data by cli, the negative timestamp doesn't show correctly
![img_v2_f7cf9fb5-b010-4cc4-89e7-9a74291daeag](https://github.com/apache/iotdb/assets/25913899/ef534d97-e589-4ccf-9dbc-bc7e121debe0)

* When insert timestamp -100, the current code will calculate the time partition id as 0. However the correct time partition id should be -1.


## How To Test

Execute the sql, check the query result is correct or not. Then check whether the time partition id is correct, which you can find in `data/datanode/data/sequence/root.sg/1`
```
insert into root.sg.d1(time,s1) values(-20,2)
select * from root.**
```
<img width="508" alt="image" src="https://github.com/apache/iotdb/assets/25913899/a461b338-9419-4cef-b4ee-f775c090659d">
